ai_services/anomaly_detection/data_preprocessing_examples/oci_data_flow_based_examples/example_code/ad_utils.py

```diff
@@ -63,28 +73,102 @@ def _create_model_(self, project_id, compartment_id, data_asset_id,
+                object_name=data_asset_detail['object'])])
+        output_details = ObjectStoreOutputDetails(
+            namespace_name=data_asset_detail['namespace'],
+            bucket_name=output_path,
+            prefix="model1")
+        create_detect_anomaly_job_details = CreateDetectAnomalyJobDetails(
+            compartment_id=compartment_id,
+            display_name="e2e_template_test_job",
+            model_id=model_id,
+            input_details=input_details,
+            output_details=output_details)
+        print(create_detect_anomaly_job_details)
+        response = self.ad_client.create_detect_anomaly_job(
+            create_detect_anomaly_job_details)
+        print(response)
+        assert response.status == 200, \
+            f"Error detecting anomalies: {response.status}"
+        return response.data
+
+    def infer(self, compartment_id:str, model_id:str, staging_details, output_path)\
+            -> int:
+        jobs = []
+        for data_asset_detail in staging_details:
+            jobs.append(self.create_detect_anomalies_job(compartment_id,
+                                                         model_id,
+                                                         data_asset_detail,
+                                                         output_path))
+        retries = 0
+        while jobs and retries < 10:
+            retries = retries + 1
+            for job in list(jobs):
+                time.sleep(self.RETRY_SECONDS)
+                response = self.ad_client.get_detect_anomaly_job(job.id)
+                if response.status != 200:
+                    f"Error fetching detect job status: {response.status}"
+                else:
+                    job = response.data
+                    if job.lifecycle_state in self.terminal_job_states:
+                        jobs.remove(job)
+                        if job.lifecycle_state != DetectAnomalyJob.LIFECYCLE_STATE_SUCCEEDED:
+                            return 1
+
+        return 0
+
 
 if __name__ == '__main__':
     parser = ArgumentParser()
     parser.add_argument("--profile_name", required=False, type=str,
                         default=DEFAULT_PROFILE)
     parser.add_argument("--service_endpoint", required=False, type=str,
                         default=None)
-    parser.add_argument("--project_id", required=True, type=str)
     parser.add_argument("--compartment_id", required=True, type=str)
-    parser.add_argument("--target_fap", required=False,
-                        type=lambda v: float(v), default=DEFAULT_TARGET_FAP)
-    parser.add_argument("--training_fraction", required=False,
-                        type=lambda v: float(v),
-                        default=DEFAULT_TRAINING_FRACTION)
-    parser.add_argument("--data_asset_detail", required=True, type=str)
+
+    subparsers = parser.add_subparsers(dest="subparser_name")
+    training_parser = subparsers.add_parser('train')
```

Comment:
[Minor] Having hard-coded values at the point of need increases the probability of human errors and leading to unexpected behavior. Please move `'train'` to a constant and use at other required places. Similar comment for other places.